### PR TITLE
Add the capability to select the arch of AT for downloading

### PR DESCRIPTION
**Pull request by rpsene**
This commit adds the capability to download AT by
	selecting a target architecture. It fixes the a
	annoying behavior related to download everything
	for a specific Linux distribution.

Signed-off-by: Rafael Peria de Sene <rpsene@br.ibm.com>

### CODE REVIEW

File: utils/at_downloader/at_downloader.sh [hunk 1]
```diff
@@ -10,14 +10,13 @@
 #   IBM Corporation, Raphael Zinsly - initial implementation and documentation.
 ###############################################################################
 #
-# The Advance Toolchain (AT) downloader is a tool to download the latest AT
-# for a supported distribution.
-# This script looks at the official FTP to find the availables distributions
-# and AT's versions.
-# The latest AT from a chosen version will be downloaded.
+# The Advance Toolchain (AT) downloader is a automated script which downloads
```
Comment (rzinsly) [inline utils/at_downloader/at_downloader.sh]: **an** automated script

File: utils/at_downloader/at_downloader.sh [hunk 2]
```diff
@@ -10,14 +10,13 @@
 #   IBM Corporation, Raphael Zinsly - initial implementation and documentation.
```
Comment (rzinsly) [inline utils/at_downloader/at_downloader.sh]: Add your name as contributor.

File: utils/at_downloader/at_downloader.sh [hunk 3]
```diff
@@ -341,36 +356,45 @@ if [[ $? -ne 0  ]]; then
...
+	# Download the packages.
+	echo "Downloading AT ${at_major}-${at_minor} on $(pwd) ..."
+	wget -A ${extensions} --progress=dot:mega ${download_url} 2>&1 \
+	| grep -E ' =>|[KM] \.| saved '
+	if [[ $? -ne 0 ]]; then
+		echo "Error: The packages were not downloaded correctly."
+		clean_exit 1
+	fi
+else
+	echo 'Could not download the required file.'
```
Comment (rzinsly) [inline utils/at_downloader/at_downloader.sh]: It's not required but would be good to print which file.

File: utils/at_downloader/at_downloader.sh [hunk 4]
```diff
@@ -341,36 +356,45 @@ if [[ $? -ne 0  ]]; then
...
+	exit 1;
```
Comment (rzinsly) [inline utils/at_downloader/at_downloader.sh]: Use the clean_exit function to clean all intermediate files.

File: utils/at_downloader/at_downloader.sh [hunk 5]
```diff
@@ -341,36 +356,45 @@ if [[ $? -ne 0  ]]; then
...
 fi
 
 # Verify the md5sum of the files downloaded.
 echo "Checking the integrity of the files..."
 # Get the list of packages and the checksums from the server.
+echo ${url}
```
Comment (rzinsly) [inline utils/at_downloader/at_downloader.sh]: Why print the url?

File: utils/at_downloader/at_downloader.sh (hunk 1 above)
Comment (rzinsly) [inline utils/at_downloader/at_downloader.sh]: Still there.

File: utils/at_downloader/at_downloader.sh (hunk 5 above)
Comment (rzinsly) [inline utils/at_downloader/at_downloader.sh]: No answer?

File: utils/at_downloader/at_downloader.sh (hunk 1 above)
Comment (rpsene) [inline utils/at_downloader/at_downloader.sh]: It shouldn't :(

File: utils/at_downloader/at_downloader.sh (hunk 5 above)
Comment (rpsene) [inline utils/at_downloader/at_downloader.sh]: It shouldn't be here!

**Pull request merged**